zerver/models.py

```diff
@@ -980,6 +985,14 @@ def allow_web_public_streams_access(self) -> bool:
         """
         return self.has_web_public_streams()
 
+    def eligible_for_zulip_communities_directory(self) -> bool:
+        if self.plan_type == Realm.PLAN_TYPE_SELF_HOSTED:
+            # Initial implementation for the Zuilip communities
+            # directory is limited to Zulip Cloud.
+            return False
+
+        return self.has_web_public_streams() or not self.invite_required
```

Comment:
I'm not sure if this is necessary now, but thought I would check before removing it and the test I wrote.

Reply:
I have since removed this function.